### PR TITLE
Ports HE piping heat glow from Bay

### PR DESCRIPTION
**Pull request by jknpj**
nicepipe
Look at the east, the pipe is burning red!

Do not merge this without #25192, seriously.

:cl:
 * rscadd: Ports HE piping heat glow from Bay

### CODE REVIEW

**Comment by brndd** [comment]
Now this is the good shit

**Comment by DamianX** [comment]
What if you made them actually glow instead of just changing color?

**Comment by jknpj** [comment]
I'm in my wagie cagie now but i'll get to it later

**Comment by jknpj** [comment]
@DamianX Here you go

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -262,3 +262,31 @@ proc/tg_jointext(list/list, glue = ",")
...
+	temp /= 100
+	if(temp <= 66)
+		. = 255
+	else
+		. = max(0, min(255, 329.698727446 * (temp - 60) ** -0.1332047592))
+
+/proc/heat2color_g(temp)
+	temp /= 100
+	if(temp <= 66)
+		. = max(0, min(255, 99.4708025861 * log(temp) - 161.1195681661))
```
Comment (Kurfursten) [inline code/__HELPERS/type2type.dm:280]: mmmm magic numbers

Comment (brndd) [reply code/__HELPERS/type2type.dm:280]: Google these numbers and you'll find a ton of results for temperature colour related code, and NONE of them actually tell you what the numbers are. These are some primordial magic numbers and nobody remembers where they originally came from or stand for.

**Comment by DamianX** [comment]
\>Do not merge this without #25192, seriously.
\>merges without #25192

**Comment by Zth--** [comment]
¯\\_(ツ)_/¯

**Pull request merged**